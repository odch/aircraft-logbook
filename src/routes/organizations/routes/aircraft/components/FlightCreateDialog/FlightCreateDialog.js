import React from 'react'
import PropTypes from 'prop-types'
import { injectIntl, intlShape, FormattedMessage } from 'react-intl'
import _get from 'lodash.get'
import Button from '@material-ui/core/Button'
import Dialog from '@material-ui/core/Dialog'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import DialogTitle from '@material-ui/core/DialogTitle'
import FormControl from '@material-ui/core/FormControl'
import FormLabel from '@material-ui/core/FormLabel'
import Grid from '@material-ui/core/Grid'
import TextField from '@material-ui/core/TextField'
import { withStyles } from '@material-ui/core/styles'
import { DatePicker, TimePicker } from 'material-ui-pickers'
import Select from '../../../../../../components/Select'
import HoursCounterField from '../../../../../../components/HoursCounterField'
import LoadingIcon from '../../../../../../components/LoadingIcon'
import {
  aerodrome as aerodromeShape,
  member as memberShape
} from '../../../../../../shapes'
import { getMemberOptions, getAerodromeOptions } from '../../util/getOptions'

const styles = {
  loadingIconContainer: {
    height: 200
  }
}

class FlightCreateDialog extends React.Component {
  handleDateChange = name => momentDate => {
    this.updateData(name, momentDate.format('YYYY-MM-DD'))
  }

  handleTimeChange = name => momentDate => {
    this.updateData(name, momentDate.format('YYYY-MM-DD HH:mm'))
  }

  handleSelectChange = name => value => {
    this.updateData(name, value)
  }

  handleHoursCounterChange = name => value => {
    this.updateData(name, value)
  }

  handleIntegerChange = name => e => {
    const value = e.target.value
    if (/^\d*$/.test(value)) {
      const intValue = value ? parseInt(value) : undefined
      this.updateData(name, intValue)
    }
  }

  updateData = (name, value) => {
    this.props.updateData({
      [name]: value
    })
  }

  getValue = (name, defaultValue) => _get(this.props.data, name, defaultValue)

  handleSubmit = e => {
    const { onSubmit, organizationId, aircraftId, data } = this.props

    e.preventDefault()
    if (onSubmit) {
      onSubmit(organizationId, aircraftId, data)
    }
  }

  msg = id => this.props.intl.formatMessage({ id })

  render() {
    const { onClose, data } = this.props
    return (
      <Dialog onClose={onClose} data-cy="flight-create-dialog" open>
        <DialogTitle>
          <FormattedMessage id="flight.create.dialog.title" />
        </DialogTitle>
        {data.initialized ? this.renderForm() : this.renderLoadingIcon()}
      </Dialog>
    )
  }

  renderForm() {
    const { onClose, organizationMembers = [], aerodromes = [] } = this.props

    const memberOptions = getMemberOptions(organizationMembers)
    const aerodromeOptions = getAerodromeOptions(aerodromes)

    return (
      <form onSubmit={this.handleSubmit}>
        <DialogContent>
          {this.renderDatePicker('date')}
          {this.renderSelect('pilot', memberOptions)}
          {this.renderSelect('departureAerodrome', aerodromeOptions)}
          {this.renderSelect('destinationAerodrome', aerodromeOptions)}
          {this.renderTimePicker('blockOffTime')}
          {this.renderTimePicker('takeOffTime')}
          {this.renderTimePicker('landingTime')}
          {this.renderTimePicker('blockOnTime')}
          <FormControl component="fieldset" margin="normal" fullWidth>
            <FormLabel component="legend">Flugstunden</FormLabel>
            <Grid container spacing={24}>
              <Grid item xs={12} sm={6}>
                {this.renderHoursCounterField('counters.flightHours.start')}
              </Grid>
              <Grid item xs={12} sm={6}>
                {this.renderHoursCounterField('counters.flightHours.end')}
              </Grid>
            </Grid>
          </FormControl>
          {this.renderNumberField('landings')}
        </DialogContent>
        <DialogActions>
          <Button onClick={onClose} color="primary">
            <FormattedMessage id="flight.create.dialog.buttons.cancel" />
          </Button>
          <Button type="submit" color="primary" data-cy="create-button">
            <FormattedMessage id="flight.create.dialog.buttons.create" />
          </Button>
        </DialogActions>
      </form>
    )
  }

  renderSelect(name, options) {
    return (
      <Select
        label={this.msg(`flight.create.dialog.${name.toLowerCase()}`)}
        value={this.props.data[name]}
        onChange={this.handleSelectChange(name)}
        options={options}
        data-cy={`${name}-field`}
        margin="normal"
      />
    )
  }

  renderDatePicker(name) {
    return (
      <DatePicker
        keyboard
        label={this.msg(`flight.create.dialog.${name.toLowerCase()}`)}
        value={this.props.data[name]}
        onChange={this.handleDateChange(name)}
        format="L"
        data-cy={`${name}-field`}
        margin="normal"
        fullWidth
        autoOk
      />
    )
  }

  renderTimePicker(name) {
    return (
      <TimePicker
        keyboard
        ampm={false}
        label={this.msg(`flight.create.dialog.${name.toLowerCase()}`)}
        value={this.props.data[name]}
        onChange={this.handleTimeChange(name)}
        data-cy={`${name}-field`}
        margin="normal"
        fullWidth
        autoOk
      />
    )
  }

  renderHoursCounterField(name) {
    return (
      <HoursCounterField
        name={name}
        label={this.msg(`flight.create.dialog.${name.toLowerCase()}`)}
        value={this.getValue(name)}
        onChange={this.handleHoursCounterChange(name)}
        cy={`${name}-field`}
        margin="normal"
        fullWidth
      />
    )
  }

  renderNumberField(name) {
    return (
      <TextField
        label={this.msg(`flight.create.dialog.${name.toLowerCase()}`)}
        value={this.getValue(name, '')}
        onChange={this.handleIntegerChange(name)}
        cy={`${name}-field`}
        type="number"
        margin="normal"
        fullWidth
      />
    )
  }

  renderLoadingIcon() {
    return (
      <div className={this.props.classes.loadingIconContainer}>
        <LoadingIcon />
      </div>
    )
  }
}

FlightCreateDialog.propTypes = {
  organizationId: PropTypes.string.isRequired,
  aircraftId: PropTypes.string.isRequired,
  data: PropTypes.shape({
    date: PropTypes.string,
    blockOffTime: PropTypes.string,
    takeOffTime: PropTypes.string,
    landingTime: PropTypes.string,
    blockOnTime: PropTypes.string,
    pilot: PropTypes.shape({
      value: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired
    }),
    departureAerodrome: PropTypes.shape({
      value: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired
    }),
    destinationAerodrome: PropTypes.shape({
      value: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired
    })
  }).isRequired,
  organizationMembers: PropTypes.arrayOf(memberShape),
  aerodromes: PropTypes.arrayOf(aerodromeShape),
  onClose: PropTypes.func,
  onSubmit: PropTypes.func,
  updateData: PropTypes.func.isRequired,
  intl: intlShape,
  classes: PropTypes.object.isRequired
}

export default injectIntl(withStyles(styles)(FlightCreateDialog))
