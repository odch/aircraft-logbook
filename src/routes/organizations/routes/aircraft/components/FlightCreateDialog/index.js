import FlightCreateDialog from './FlightCreateDialog'

export default FlightCreateDialog
