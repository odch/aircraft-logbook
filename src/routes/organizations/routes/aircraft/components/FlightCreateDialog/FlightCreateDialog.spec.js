import React from 'react'
import { MuiPickersUtilsProvider } from 'material-ui-pickers'
import MomentUtils from '@date-io/moment'
import { renderIntlMaterial } from '../../../../../../testutil/renderIntl'
import FlightCreateDialog from './FlightCreateDialog'

describe('routes', () => {
  describe('organizations', () => {
    describe('routes', () => {
      describe('aircraft', () => {
        describe('components', () => {
          describe('FlightCreateDialog', () => {
            it('renders correctly', () => {
              const renderedValue = renderIntlMaterial(
                <MuiPickersUtilsProvider utils={MomentUtils}>
                  <FlightCreateDialog
                    organizationId="my_org"
                    aircraftId="o7flC7jw8jmkOfWo8oyA"
                    data={{
                      initialized: true,
                      date: '2018-11-30',
                      blockOffTime: '2018-11-30 16:00',
                      takeOffTime: '2018-11-30 16:10',
                      landingTime: '2018-11-30 16:30',
                      blockOnTime: '2018-11-30 16:40'
                    }}
                    updateData={() => {}}
                  />
                </MuiPickersUtilsProvider>,
                true
              )
              expect(renderedValue).toMatchSnapshot()
            })

            it('renders disabled when submitting', () => {
              const renderedValue = renderIntlMaterial(
                <MuiPickersUtilsProvider utils={MomentUtils}>
                  <FlightCreateDialog
                    organizationId="my_org"
                    aircraftId="o7flC7jw8jmkOfWo8oyA"
                    data={{
                      initialized: true,
                      date: '2018-11-30',
                      blockOffTime: '2018-11-30 16:00',
                      takeOffTime: '2018-11-30 16:10',
                      landingTime: '2018-11-30 16:30',
                      blockOnTime: '2018-11-30 16:40'
                    }}
                    submitting
                    updateData={() => {}}
                  />
                </MuiPickersUtilsProvider>,
                true
              )
              expect(renderedValue).toMatchSnapshot()
            })

            it('renders readonly fields disabled', () => {
              const renderedValue = renderIntlMaterial(
                <MuiPickersUtilsProvider utils={MomentUtils}>
                  <FlightCreateDialog
                    organizationId="my_org"
                    aircraftId="o7flC7jw8jmkOfWo8oyA"
                    data={{
                      initialized: true,
                      date: '2018-11-30',
                      blockOffTime: '2018-11-30 16:00',
                      takeOffTime: '2018-11-30 16:10',
                      landingTime: '2018-11-30 16:30',
                      blockOnTime: '2018-11-30 16:40'
                    }}
                    readOnlyFields={['date', 'blockOffTime']}
                    updateData={() => {}}
                  />
                </MuiPickersUtilsProvider>,
                true
              )
              expect(renderedValue).toMatchSnapshot()
            })
          })
        })
      })
    })
  })
})
