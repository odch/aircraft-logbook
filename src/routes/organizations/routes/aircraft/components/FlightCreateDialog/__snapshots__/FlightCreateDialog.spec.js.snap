// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes organizations routes aircraft components FlightCreateDialog renders correctly 1`] = `
"<div class=\\"MuiModal-root-14 MuiDialog-root-2 MuiDialog-scrollPaper-3\\" role=\\"dialog\\" data-cy=\\"flight-create-dialog\\">
  <div class=\\"MuiBackdrop-root-16\\" aria-hidden=\\"true\\" style=\\"opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div>
  <div class=\\"MuiPaper-root-18 MuiPaper-elevation24-44 MuiPaper-rounded-19 MuiDialog-paper-5 MuiDialog-paperScrollPaper-6 MuiDialog-paperWidthSm-9\\" style=\\"opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"document\\" tabindex=\\"-1\\">
    <div class=\\"MuiDialogTitle-root-45\\">
      <h2 class=\\"MuiTypography-root-46 MuiTypography-title-52\\"><span>Flug erfassen</span></h2>
    </div>
    <form>
      <div class=\\"MuiDialogContent-root-72\\">
        <div class=\\"MuiFormControl-root-73 MuiFormControl-marginNormal-74 MuiFormControl-fullWidth-76\\" data-cy=\\"date-field\\"><label class=\\"MuiFormLabel-root-84 MuiFormLabel-filled-88 MuiInputLabel-root-77 MuiInputLabel-formControl-78 MuiInputLabel-animated-81 MuiInputLabel-shrink-80\\" data-shrink=\\"true\\">Datum</label>
          <div class=\\"MuiInputBase-root-104 MuiInput-root-91 MuiInput-underline-95 MuiInputBase-fullWidth-113 MuiInput-fullWidth-98 MuiInputBase-formControl-105 MuiInput-formControl-92 MuiInputBase-adornedEnd-109\\"><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input-114 MuiInput-input-99 MuiInputBase-inputAdornedEnd-120\\" type=\\"text\\" value=\\"30.11.2018\\">
            <div class=\\"MuiInputAdornment-root-121 MuiInputAdornment-positionEnd-124\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-131 MuiIconButton-root-125\\" type=\\"button\\"><span class=\\"MuiIconButton-label-130\\"><span class=\\"material-icons MuiIcon-root-134\\" aria-hidden=\\"true\\"> event </span></span><span class=\\"MuiTouchRipple-root-143\\"></span></button></div>
          </div>
        </div>
        <div class=\\"IntegrationReactSelect-root-157\\">
          <div class=\\"css-10nd86i\\">
            <div class=\\"MuiFormControl-root-73 MuiFormControl-marginNormal-74 MuiFormControl-fullWidth-76\\"><label class=\\"MuiFormLabel-root-84 MuiInputLabel-root-77 MuiInputLabel-formControl-78 MuiInputLabel-animated-81 MuiInputLabel-shrink-80\\" data-shrink=\\"true\\">Pilot</label>
              <div class=\\"MuiInputBase-root-104 MuiInput-root-91 MuiInput-underline-95 MuiInputBase-fullWidth-113 MuiInput-fullWidth-98 MuiInputBase-formControl-105 MuiInput-formControl-92\\">
                <div aria-invalid=\\"false\\" class=\\"MuiInputBase-input-114 MuiInput-input-99 IntegrationReactSelect-input-158\\" type=\\"text\\">
                  <div class=\\"IntegrationReactSelect-valueContainer-159\\">
                    <p class=\\"MuiTypography-root-46 MuiTypography-body1-55 MuiTypography-colorTextSecondary-70 IntegrationReactSelect-placeholder-164\\"></p>
                    <div class=\\"css-13fspk8\\">
                      <div class=\\"\\" style=\\"display: inline-block;\\"><input autocapitalize=\\"none\\" autocomplete=\\"off\\" autocorrect=\\"off\\" id=\\"react-select-2-input\\" spellcheck=\\"false\\" tabindex=\\"0\\" type=\\"text\\" aria-autocomplete=\\"list\\" style=\\"box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;\\" value=\\"\\">
                        <div style=\\"position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; letter-spacing: normal; text-transform: none;\\"></div>
                      </div>
                    </div>
                  </div>
                  <div class=\\"css-1wy0on6\\"><span class=\\"css-d8oujb\\"></span>
                    <div aria-hidden=\\"true\\" class=\\"css-fyho0h\\"><svg height=\\"20\\" width=\\"20\\" viewBox=\\"0 0 20 20\\" aria-hidden=\\"true\\" focusable=\\"false\\" class=\\"css-19bqh2r\\">
                        <path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\"></path>
                      </svg></div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class=\\"IntegrationReactSelect-root-157\\">
          <div class=\\"css-10nd86i\\">
            <div class=\\"MuiFormControl-root-73 MuiFormControl-marginNormal-74 MuiFormControl-fullWidth-76\\"><label class=\\"MuiFormLabel-root-84 MuiInputLabel-root-77 MuiInputLabel-formControl-78 MuiInputLabel-animated-81 MuiInputLabel-shrink-80\\" data-shrink=\\"true\\">Typ</label>
              <div class=\\"MuiInputBase-root-104 MuiInput-root-91 MuiInput-underline-95 MuiInputBase-fullWidth-113 MuiInput-fullWidth-98 MuiInputBase-formControl-105 MuiInput-formControl-92\\">
                <div aria-invalid=\\"false\\" class=\\"MuiInputBase-input-114 MuiInput-input-99 IntegrationReactSelect-input-158\\" type=\\"text\\">
                  <div class=\\"IntegrationReactSelect-valueContainer-159\\">
                    <p class=\\"MuiTypography-root-46 MuiTypography-body1-55 MuiTypography-colorTextSecondary-70 IntegrationReactSelect-placeholder-164\\"></p>
                    <div class=\\"css-13fspk8\\">
                      <div class=\\"\\" style=\\"display: inline-block;\\"><input autocapitalize=\\"none\\" autocomplete=\\"off\\" autocorrect=\\"off\\" id=\\"react-select-3-input\\" spellcheck=\\"false\\" tabindex=\\"0\\" type=\\"text\\" aria-autocomplete=\\"list\\" style=\\"box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;\\" value=\\"\\">
                        <div style=\\"position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; letter-spacing: normal; text-transform: none;\\"></div>
                      </div>
                    </div>
                  </div>
                  <div class=\\"css-1wy0on6\\"><span class=\\"css-d8oujb\\"></span>
                    <div aria-hidden=\\"true\\" class=\\"css-fyho0h\\"><svg height=\\"20\\" width=\\"20\\" viewBox=\\"0 0 20 20\\" aria-hidden=\\"true\\" focusable=\\"false\\" class=\\"css-19bqh2r\\">
                        <path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\"></path>
                      </svg></div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class=\\"IntegrationReactSelect-root-157\\">
          <div class=\\"css-10nd86i\\">
            <div class=\\"MuiFormControl-root-73 MuiFormControl-marginNormal-74 MuiFormControl-fullWidth-76\\"><label class=\\"MuiFormLabel-root-84 MuiInputLabel-root-77 MuiInputLabel-formControl-78 MuiInputLabel-animated-81 MuiInputLabel-shrink-80\\" data-shrink=\\"true\\">Startflugplatz</label>
              <div class=\\"MuiInputBase-root-104 MuiInput-root-91 MuiInput-underline-95 MuiInputBase-fullWidth-113 MuiInput-fullWidth-98 MuiInputBase-formControl-105 MuiInput-formControl-92\\">
                <div aria-invalid=\\"false\\" class=\\"MuiInputBase-input-114 MuiInput-input-99 IntegrationReactSelect-input-158\\" type=\\"text\\">
                  <div class=\\"IntegrationReactSelect-valueContainer-159\\">
                    <p class=\\"MuiTypography-root-46 MuiTypography-body1-55 MuiTypography-colorTextSecondary-70 IntegrationReactSelect-placeholder-164\\"></p>
                    <div class=\\"css-13fspk8\\">
                      <div class=\\"\\" style=\\"display: inline-block;\\"><input autocapitalize=\\"none\\" autocomplete=\\"off\\" autocorrect=\\"off\\" id=\\"react-select-4-input\\" spellcheck=\\"false\\" tabindex=\\"0\\" type=\\"text\\" aria-autocomplete=\\"list\\" style=\\"box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;\\" value=\\"\\">
                        <div style=\\"position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; letter-spacing: normal; text-transform: none;\\"></div>
                      </div>
                    </div>
                  </div>
                  <div class=\\"css-1wy0on6\\"><span class=\\"css-d8oujb\\"></span>
                    <div aria-hidden=\\"true\\" class=\\"css-fyho0h\\"><svg height=\\"20\\" width=\\"20\\" viewBox=\\"0 0 20 20\\" aria-hidden=\\"true\\" focusable=\\"false\\" class=\\"css-19bqh2r\\">
                        <path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\"></path>
                      </svg></div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class=\\"IntegrationReactSelect-root-157\\">
          <div class=\\"css-10nd86i\\">
            <div class=\\"MuiFormControl-root-73 MuiFormControl-marginNormal-74 MuiFormControl-fullWidth-76\\"><label class=\\"MuiFormLabel-root-84 MuiInputLabel-root-77 MuiInputLabel-formControl-78 MuiInputLabel-animated-81 MuiInputLabel-shrink-80\\" data-shrink=\\"true\\">Zielflugplatz</label>
              <div class=\\"MuiInputBase-root-104 MuiInput-root-91 MuiInput-underline-95 MuiInputBase-fullWidth-113 MuiInput-fullWidth-98 MuiInputBase-formControl-105 MuiInput-formControl-92\\">
                <div aria-invalid=\\"false\\" class=\\"MuiInputBase-input-114 MuiInput-input-99 IntegrationReactSelect-input-158\\" type=\\"text\\">
                  <div class=\\"IntegrationReactSelect-valueContainer-159\\">
                    <p class=\\"MuiTypography-root-46 MuiTypography-body1-55 MuiTypography-colorTextSecondary-70 IntegrationReactSelect-placeholder-164\\"></p>
                    <div class=\\"css-13fspk8\\">
                      <div class=\\"\\" style=\\"display: inline-block;\\"><input autocapitalize=\\"none\\" autocomplete=\\"off\\" autocorrect=\\"off\\" id=\\"react-select-5-input\\" spellcheck=\\"false\\" tabindex=\\"0\\" type=\\"text\\" aria-autocomplete=\\"list\\" style=\\"box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;\\" value=\\"\\">
                        <div style=\\"position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; letter-spacing: normal; text-transform: none;\\"></div>
                      </div>
                    </div>
                  </div>
                  <div class=\\"css-1wy0on6\\"><span class=\\"css-d8oujb\\"></span>
                    <div aria-hidden=\\"true\\" class=\\"css-fyho0h\\"><svg height=\\"20\\" width=\\"20\\" viewBox=\\"0 0 20 20\\" aria-hidden=\\"true\\" focusable=\\"false\\" class=\\"css-19bqh2r\\">
                        <path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\"></path>
                      </svg></div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class=\\"MuiFormControl-root-73 MuiFormControl-marginNormal-74 MuiFormControl-fullWidth-76\\" data-cy=\\"blockOffTime-field\\"><label class=\\"MuiFormLabel-root-84 MuiFormLabel-filled-88 MuiInputLabel-root-77 MuiInputLabel-formControl-78 MuiInputLabel-animated-81 MuiInputLabel-shrink-80\\" data-shrink=\\"true\\">Block off</label>
          <div class=\\"MuiInputBase-root-104 MuiInput-root-91 MuiInput-underline-95 MuiInputBase-fullWidth-113 MuiInput-fullWidth-98 MuiInputBase-formControl-105 MuiInput-formControl-92 MuiInputBase-adornedEnd-109\\"><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input-114 MuiInput-input-99 MuiInputBase-inputAdornedEnd-120\\" type=\\"text\\" value=\\"16:00\\">
            <div class=\\"MuiInputAdornment-root-121 MuiInputAdornment-positionEnd-124\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-131 MuiIconButton-root-125\\" type=\\"button\\"><span class=\\"MuiIconButton-label-130\\"><span class=\\"material-icons MuiIcon-root-134\\" aria-hidden=\\"true\\"> event </span></span><span class=\\"MuiTouchRipple-root-143\\"></span></button></div>
          </div>
        </div>
        <div class=\\"MuiFormControl-root-73 MuiFormControl-marginNormal-74 MuiFormControl-fullWidth-76\\" data-cy=\\"takeOffTime-field\\"><label class=\\"MuiFormLabel-root-84 MuiFormLabel-filled-88 MuiInputLabel-root-77 MuiInputLabel-formControl-78 MuiInputLabel-animated-81 MuiInputLabel-shrink-80\\" data-shrink=\\"true\\">Take off</label>
          <div class=\\"MuiInputBase-root-104 MuiInput-root-91 MuiInput-underline-95 MuiInputBase-fullWidth-113 MuiInput-fullWidth-98 MuiInputBase-formControl-105 MuiInput-formControl-92 MuiInputBase-adornedEnd-109\\"><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input-114 MuiInput-input-99 MuiInputBase-inputAdornedEnd-120\\" type=\\"text\\" value=\\"16:10\\">
            <div class=\\"MuiInputAdornment-root-121 MuiInputAdornment-positionEnd-124\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-131 MuiIconButton-root-125\\" type=\\"button\\"><span class=\\"MuiIconButton-label-130\\"><span class=\\"material-icons MuiIcon-root-134\\" aria-hidden=\\"true\\"> event </span></span><span class=\\"MuiTouchRipple-root-143\\"></span></button></div>
          </div>
        </div>
        <div class=\\"MuiFormControl-root-73 MuiFormControl-marginNormal-74 MuiFormControl-fullWidth-76\\" data-cy=\\"landingTime-field\\"><label class=\\"MuiFormLabel-root-84 MuiFormLabel-filled-88 MuiInputLabel-root-77 MuiInputLabel-formControl-78 MuiInputLabel-animated-81 MuiInputLabel-shrink-80\\" data-shrink=\\"true\\">Landung</label>
          <div class=\\"MuiInputBase-root-104 MuiInput-root-91 MuiInput-underline-95 MuiInputBase-fullWidth-113 MuiInput-fullWidth-98 MuiInputBase-formControl-105 MuiInput-formControl-92 MuiInputBase-adornedEnd-109\\"><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input-114 MuiInput-input-99 MuiInputBase-inputAdornedEnd-120\\" type=\\"text\\" value=\\"16:30\\">
            <div class=\\"MuiInputAdornment-root-121 MuiInputAdornment-positionEnd-124\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-131 MuiIconButton-root-125\\" type=\\"button\\"><span class=\\"MuiIconButton-label-130\\"><span class=\\"material-icons MuiIcon-root-134\\" aria-hidden=\\"true\\"> event </span></span><span class=\\"MuiTouchRipple-root-143\\"></span></button></div>
          </div>
        </div>
        <div class=\\"MuiFormControl-root-73 MuiFormControl-marginNormal-74 MuiFormControl-fullWidth-76\\" data-cy=\\"blockOnTime-field\\"><label class=\\"MuiFormLabel-root-84 MuiFormLabel-filled-88 MuiInputLabel-root-77 MuiInputLabel-formControl-78 MuiInputLabel-animated-81 MuiInputLabel-shrink-80\\" data-shrink=\\"true\\">Block on</label>
          <div class=\\"MuiInputBase-root-104 MuiInput-root-91 MuiInput-underline-95 MuiInputBase-fullWidth-113 MuiInput-fullWidth-98 MuiInputBase-formControl-105 MuiInput-formControl-92 MuiInputBase-adornedEnd-109\\"><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input-114 MuiInput-input-99 MuiInputBase-inputAdornedEnd-120\\" type=\\"text\\" value=\\"16:40\\">
            <div class=\\"MuiInputAdornment-root-121 MuiInputAdornment-positionEnd-124\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-131 MuiIconButton-root-125\\" type=\\"button\\"><span class=\\"MuiIconButton-label-130\\"><span class=\\"material-icons MuiIcon-root-134\\" aria-hidden=\\"true\\"> event </span></span><span class=\\"MuiTouchRipple-root-143\\"></span></button></div>
          </div>
        </div>
        <fieldset class=\\"MuiFormControl-root-73 MuiFormControl-marginNormal-74 MuiFormControl-fullWidth-76\\">
          <legend class=\\"MuiFormLabel-root-84\\">Flugstunden</legend>
          <div class=\\"MuiGrid-container-167 MuiGrid-spacing-xs-24-191\\">
            <div class=\\"MuiGrid-item-168 MuiGrid-grid-xs-12-207 MuiGrid-grid-sm-6-215\\">
              <div class=\\"MuiFormControl-root-73 MuiFormControl-marginNormal-74 MuiFormControl-fullWidth-76\\" data-cy=\\"counters.flightHours.start-field\\"><label class=\\"MuiFormLabel-root-84 MuiInputLabel-root-77 MuiInputLabel-formControl-78 MuiInputLabel-animated-81\\" data-shrink=\\"false\\">Startzähler</label>
                <div class=\\"MuiInputBase-root-104 MuiInput-root-91 MuiInput-underline-95 MuiInputBase-fullWidth-113 MuiInput-fullWidth-98 MuiInputBase-formControl-105 MuiInput-formControl-92\\"><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input-114 MuiInput-input-99 MuiInputBase-inputType-117 MuiInput-inputType-102\\" type=\\"number\\" step=\\".01\\" value=\\"\\"></div>
              </div>
            </div>
            <div class=\\"MuiGrid-item-168 MuiGrid-grid-xs-12-207 MuiGrid-grid-sm-6-215\\">
              <div class=\\"MuiFormControl-root-73 MuiFormControl-marginNormal-74 MuiFormControl-fullWidth-76\\" data-cy=\\"counters.flightHours.end-field\\"><label class=\\"MuiFormLabel-root-84 MuiInputLabel-root-77 MuiInputLabel-formControl-78 MuiInputLabel-animated-81\\" data-shrink=\\"false\\">Endzähler</label>
                <div class=\\"MuiInputBase-root-104 MuiInput-root-91 MuiInput-underline-95 MuiInputBase-fullWidth-113 MuiInput-fullWidth-98 MuiInputBase-formControl-105 MuiInput-formControl-92\\"><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input-114 MuiInput-input-99 MuiInputBase-inputType-117 MuiInput-inputType-102\\" type=\\"number\\" step=\\".01\\" value=\\"\\"></div>
              </div>
            </div>
          </div>
        </fieldset>
        <div class=\\"MuiFormControl-root-73 MuiFormControl-marginNormal-74 MuiFormControl-fullWidth-76\\" cy=\\"landings-field\\"><label class=\\"MuiFormLabel-root-84 MuiInputLabel-root-77 MuiInputLabel-formControl-78 MuiInputLabel-animated-81\\" data-shrink=\\"false\\">Landungen</label>
          <div class=\\"MuiInputBase-root-104 MuiInput-root-91 MuiInput-underline-95 MuiInputBase-fullWidth-113 MuiInput-fullWidth-98 MuiInputBase-formControl-105 MuiInput-formControl-92\\"><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input-114 MuiInput-input-99 MuiInputBase-inputType-117 MuiInput-inputType-102\\" type=\\"number\\" value=\\"\\"></div>
        </div>
      </div>
      <div class=\\"MuiDialogActions-root-264\\"><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-131 MuiButton-root-266 MuiButton-text-268 MuiButton-textPrimary-269 MuiButton-flat-271 MuiButton-flatPrimary-272 MuiDialogActions-action-265\\" type=\\"button\\"><span class=\\"MuiButton-label-267\\"><span>Abbrechen</span></span><span class=\\"MuiTouchRipple-root-143\\"></span></button><button tabindex=\\"0\\" class=\\"MuiButtonBase-root-131 MuiButton-root-266 MuiButton-text-268 MuiButton-textPrimary-269 MuiButton-flat-271 MuiButton-flatPrimary-272 MuiDialogActions-action-265\\" type=\\"submit\\" data-cy=\\"create-button\\"><span class=\\"MuiButton-label-267\\"><span>Speichern</span></span><span class=\\"MuiTouchRipple-root-143\\"></span></button></div>
    </form>
  </div>
</div>"
`;
